Sources/Workspace/UserToolchain.swift

```diff
@@ -302,19 +307,31 @@ public final class UserToolchain: Toolchain {
 
     // MARK: - initializer
 
-    public init(destination: Destination, environment: EnvironmentVariables = .process()) throws {
+    public enum SearchStrategy {
+        case `default`
+        case custom(searchPaths: [AbsolutePath], useXcrun: Bool = true)
```

Comment:
I played around with having a different `custom` based on platform, but I think that makes the code much harder to read for little benefit.